acceptance/suites/puppet3_tests/020_backwards_compat/test.rb

```diff
@@ -0,0 +1,118 @@
+# require 'test/unit/assertions'
+# require 'puppet/acceptance/module_utils'
+
+test_name "3.x agent running against 4.x master"
+
+studio = "/tmp/simmons-studio-#{Process.pid}"
+
+teardown do
+  on(agents, "rm -rf #{studio}")
+end
+
+step "Install modules" do
+  simmons_version = '0.2.0'
+  on(master, puppet("module install nwolfe-simmons --version #{simmons_version}"))
+  on(master, puppet("module list"))
+end
+
+step "Configure file serving" do
+  fileserverconf = on(master, puppet("config print fileserverconfig")).stdout.chomp
+  create_remote_file(master, fileserverconf, <<FILESERVERCONF)
+[simmons_custom_mount_point]
+path /etc/puppetlabs/code/environments/production/modules/simmons/mount-point-files
+allow *
+FILESERVERCONF
+  on(master, "chmod 644 #{fileserverconf}")
+end
+
+step "Disable auth.conf" do
+  authconf = on(master, puppet("config print rest_authconfig")).stdout.chomp
+  create_remote_file(master, authconf, <<AUTHCONF)
+path /
+auth any
+allow *
+AUTHCONF
+  on(master, "chmod 644 #{authconf}")
+end
+
+step "Perform agent upgrade steps" do
+  step "Enable structured facts" do
+    agents.each do |agent|
+      on(agent, puppet("config set stringify_facts false --section agent"))
+    end
+  end
+end
+
+step "Configure site.pp" do
+  sitepp = '/etc/puppetlabs/code/environments/production/manifests/site.pp'
+  create_remote_file(master, sitepp, <<SITEPP)
+class { 'simmons':
+  studio => "#{studio}",
+}
+SITEPP
+  on(master, "chmod 644 #{sitepp}")
+end
+
+agent = master
+
+step "Run agents" do
+  with_puppet_running_on(master, {}) do
```

Comment:
When this first Puppet agent run occurs, it should have to negotiate with the master to get a certificate - agent submits a cert request, master signs the cert, agent gets the signed cert from the master - right?  Does the master need to configured for { "autosign" => "true" } in order for the cert to be signed automatically.  Looks like we do that along with setting some other properties for the master in the `with_puppet_running_on` call which happens from `puppetserver_initialize_ssl` from `.../acceptance/lib/helper.rb`, for example.